lib/TcpConnection.js

```diff
@@ -84,6 +86,10 @@ module.exports = function factory (userOptions) {
       if (socket.writableEnded) {
         handleSocketWriteError(new Error('This socket has been ended by the other party'), data, encoding)
       } else {
+        if (options.onBeforeDataWrite) {
+          data = options.onBeforeDataWrite(data)
```

Comment:
Not sure if this is bad practice here, but I've overwritten the existing data value to spare memory overhead from having to create another copy of the data.

If you're not into that, I can do something of the following:

```
let newData

if (options.onBeforeDataWrite) {
  newData = options.onBeforeDataWrite(data)
}

socket.write(newData | data, ...)
```